das/bigcache_storage_service.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"context"
+	"encoding/base32"
+	"fmt"
+	"time"
+
+	"github.com/allegro/bigcache"
+	flag "github.com/spf13/pflag"
+)
+
+type BigCacheConfig struct {
+	// TODO add other config information like HardMaxCacheSize
+	Expiration time.Duration `koanf:"big-cache-expiration"`
+}
+
+var DefaultBigCacheConfig = BigCacheConfig{
+	Expiration: time.Hour,
+}
+
+func BigCacheConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.Duration(prefix+".big-cache-expiration", DefaultBigCacheConfig.Expiration, "Big cache expiration")
+}
+
+type BigCacheStorageService struct {
+	baseStorageService StorageService
+	bigCacheConfig     BigCacheConfig
+	bigCache           *bigcache.BigCache
+}
+
+func NewBigCacheStorageService(bigCacheConfig BigCacheConfig, baseStorageService StorageService) (StorageService, error) {
+	bigCache, err := bigcache.NewBigCache(bigcache.DefaultConfig(10 * time.Minute))
+	if err != nil {
+		return nil, err
+	}
+	return &BigCacheStorageService{
+		baseStorageService: baseStorageService,
+		bigCacheConfig:     bigCacheConfig,
+		bigCache:           bigCache,
+	}, nil
+}
+
+func (bcs *BigCacheStorageService) GetByHash(ctx context.Context, key []byte) ([]byte, error) {
+	ret, err := bcs.bigCache.Get(base32.StdEncoding.EncodeToString(key))
```

Comment:
Do we need to do this encoding, or can we use `key` directly?  For the filesystem-backed implementation, we need to encode to make sure we have a valid file name. Unless BigCache has a similar requirement, it seems better to use the key directly.